plugins/search-common/src/types.ts

```diff
@@ -28,12 +28,34 @@ export interface SearchQuery {
   pageCursor?: string;
 }
 
+/**
+ * @beta
+ */
+export interface ResultHighlight {
```

Comment:
Might be missing something, but I don't find this data format to be super intuitive, and seems to require a bit of processing to be able to display the highlights too. I wonder if something where we do a bit more processing in the backend could make sense? And perhaps have the highlights expressed as text offsets with metadata.

Reply:
hm the motivation behind this is that both PG and ES produces highlighted results by wrapping matched terms in tags. It seemed more straightforward to supply this directly to the frontend to process via simple find/replace. If we choose to transport this metadata via offsets then additional processing is required to both "encode" on the backend and "decode" on the frontend. Offset math also introduces more complexity on both ends IMO.

Reply:
Yeah, given ES is going to probably be the workhorse search engine, it seems reasonable to mirror our approach to the format it's settled on.  FWIW, [Algolia's API responses](https://www.algolia.com/doc/api-reference/api-methods/search/#json-format) have similar assumptions.  [Solr as well](https://solr.apache.org/guide/6_6/highlighting.html#Highlighting-HighlightingintheQueryResponse).

Reply:
@Rugvip updated!